frontend/src/lib/query.js

```diff
@@ -390,6 +393,40 @@ var Query = {
         }
     },
 
+    addExpression(query) {
+        let expressions = query.expressions || {};
+        expressions[''] = [null, ''];
+        query.expressions = expressions;
+
+        console.log('added expressions. expressions =', query.expressions);
+    },
+
+    updateExpression(query, name, expression) {
+        query.expressions[name] = expression;
+
+        console.log('updated expression. expressions =', query.expressions);
+    },
+
+    renameExpression(query, oldName, newName) {
+        if (oldName === newName) return;
+
+        oldName = oldName || '';
+        query.expressions[newName] = query.expressions[oldName];
+        delete query.expressions[oldName];
+
+        console.log('renamed expression. expressions =', query.expressions);
+    },
+
+    removeExpression(query, name) {
+        if (!query.expressions) return;
+
+        delete query.expressions[name];
+
+        if (_.isEmpty(query.expressions)) delete query.expressions;
+
+        console.log('removed expression. expressions =', query.expressions);
+    },
```

Comment:
AggregationClause looks like it has a handful of methods for operating specifically on the `AggregationClause` itself but the logic for operating on a query WRE the aggregation clause is still part of `Query` (`getAggregationDescription`, `getAggregationType`, `getAggregationField`, `hasValidAggregation`, `hasEmptyAggregation`, `isBareRowsAggregation`, and `updateAggregation` are all part of `Query`). So I think it makes sense to follow that pattern and keep functions like `updateExpression` in `Query`.